crates/controller/src/hud/minimap/interaction.rs

```diff
@@ -63,50 +76,117 @@ impl MinimapClickEvent {
     }
 }
 
-impl fmt::Debug for MinimapClickEvent {
+impl fmt::Debug for MinimapPressEvent {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         write!(f, "{:?} -> {:?}", self.button, self.position)
     }
 }
 
+#[derive(Event)]
+struct MinimapDragEvent {
+    button: MouseButton,
+    position: Vec2,
+}
+
+impl MinimapDragEvent {
+    fn new(button: MouseButton, position: Vec2) -> Self {
+        Self { button, position }
+    }
+
+    fn button(&self) -> MouseButton {
+        self.button
+    }
+
+    /// Position on the map in 2D flat coordinates (these are not minimap
+    /// coordinates).
+    fn position(&self) -> Vec2 {
+        self.position
+    }
+}
+
+#[derive(Resource)]
+struct DraggingButtons(Vec<MouseButton>);
+
 fn click_handler(
     window_query: Query<&Window, With<PrimaryWindow>>,
     mut input_events: EventReader<MouseButtonInput>,
     hud: HudNodes<With<MinimapNode>>,
     bounds: Res<MapBounds>,
-    mut click_events: EventWriter<MinimapClickEvent>,
+    mut dragging: ResMut<DraggingButtons>,
+    mut click_events: EventWriter<MinimapPressEvent>,
 ) {
     let Some(cursor) = window_query.single().cursor_position() else {
         return;
     };
+
     for event in input_events.iter() {
-        if event.state != ButtonState::Released {
+        if event.state != ButtonState::Pressed {
+            dragging.0.retain(|b| *b != event.button);
```

Comment:
This must be done before the early return (let-else above), otherwise the button would stay "pressed" indefinitely if released outside of the game window.

Also, please test that it works correctly when:

1. press the button while in the minmap
1. hold the button and switch to a different window (e.g. with Alt-TAB)
1. release the button
1. switch back to the game

Reply:
At least on Windows, when I click -> alt-tab -> release, I don't seem to be receiving a mouse released event, unless the mouse is over the window.
On the other hand when I simply drag from the window to the outside, I consistently receive a release event, but `cursor_position` is inconsistently sometimes `None` and sometimes `Some` with the correct coordinates outside the window.
Any tricks for this?

Reply:
What about checking current state (i.e. Bevy's `Input<MouseButton>::.pressed(MouseButton::Left)`) & combining that with flag whether the last press event occurred over the minimap (every time the mouse button is click, we would update en enum of where it was, i.e. `LastPressLocation::Minimap`)?

Reply:
`Input<MouseButton>` ends up having the same shortcomings, as it is internally being updated using the same `MouseButtonInput` events. Seems like we almost need global input if we want to handle this case "perfectly".

Reply:
I am ok with merging it as is. Can you please open a new issue so that we fix that later?

Reply:
I was thinking the same.